R/build-logo.R

```diff
@@ -1,29 +1,113 @@
 build_logo <- function(pkg = ".") {
+
   pkg <- as_pkgdown(pkg)
 
-  logo_path <- find_logo(pkg$src_path)
-  if (is.null(logo_path))
+  if (!dir_exists(path(pkg$src_path, "pkgdown", "favicon"))) {
     return()
+  }
+
+  dir_copy_to(pkg, path(pkg$src_path, "pkgdown", "favicon"), pkg$dst_path)
 
-  file_copy_to(pkg, logo_path, from_dir = path_dir(logo_path))
+}
 
-  if (!requireNamespace("magick", quietly = TRUE)) {
-    message("magick not available, using default pkgdown favicon.ico")
 
-    if (!file_exists(path(pkg$dst_path, "favicon.ico"))) {
-      file_copy(path_pkgdown("assets", "favicon.ico"), pkg$dst_path)
-    }
-    return()
+#' Create a complete set of favicons with different sizes from your package logo
+#'
+#' This function auto-detects the location of your package logo (with the name
+#' `logo.csv` (recommended format) or `logo.png`) and runs it through the
+#' <https://realfavicongenerator.net> API to build a complete set of favicons
+#' with different sizes, as needed for modern web usage.
+#'
+#' You only need to run the function once. The favicon set will be stored in
+#' `pkgdown/favicon` and copied by [init_site()] to the relevant location each
+#' time the website is rebuilt.
+#'
+#' @inheritParams as_pkgdown
+#'
+#' @export
+build_favicon <- function(pkg = ".") {
+
+  pkg <- as_pkgdown(pkg)
+
+  logo_path <- find_logo(pkg$src_path)
+
+  if (is.null(logo_path)) {
+    stop("Package logo could not be found. Aborting favicon creation.",
+         call. = FALSE)
   }
 
-  cat_line("Creating ", dst_path("favicon.ico"))
-  magick::image_read(logo_path) %>%
-	  magick::image_scale("32x32") %>%
-	  magick::image_write(path(pkg$dst_path, "favicon.ico"), format = "png")
+  message("Creating favicon set.")
```

Comment:
I think you can remove this message, since the user will have just called `build_favicon()`

Reply:
Given this function can be a bit long, it could be useful to have some visual feedback to tell the users that it is running and it did not simply freeze. 

I just had a look at how to use `cli` spinners. What I don't understand is that they are apparently already used in this package (e.g. at the beginning of the output of `build_site()`) but I could not locate the code controlling it.

Reply:
I don't think you can easily do them with httr currently.